### ISSUE TITLE
Creating installation Live ISO with EFI support 

### ISSUE THREAD

**Issue by asharrem** [mvallim/live-custom-ubuntu-from-scratch#32]
Testing builds on Virtualbox with EFI enabled - ISO fails to run.
With EFI off, ubiquity crashes likely at Grub install stage.
Target system is "bionic".

During build process, grub-pc does not run?
Presume grub-install needs to run during build process.
May missing something from chroot_build like;
grub-install --target=x86_64-efi --efi-directory=/boot/efi --recheck --no-nvram --removable

Will test & update...

**Comment by mvallim** [comment]
Hi @asharrem,

_Testing builds on Virtualbox with EFI enabled - ISO fails to run.
With EFI off, ubiquity crashes likely at Grub install stage.
Target system is "bionic"._

**I cannot reproduce this error.
Can you give more details of the entire process you are running?
What is your OS? Do you use UEFI on the host?**

_During build process, grub-pc does not run?
Presume grub-install needs to run during build process.
May missing something from chroot_build like;
grub-install --target=x86_64-efi --efi-directory=/boot/efi --recheck --no-nvram --removable_

**Not, necessary.**

Thanks for all,

**Comment by mvallim** [comment]
Hi @asharrem,

I've made some niceties to the documentation and imaging scripts.

Could you check if everything is OK?

Thank you very much,

**Comment by asharrem** [comment]
Hi @mvallim
First of all...Thank you so very much for your wonderful script. I have used so many different methods to build/customise a live & installable ISO, this is the most ideal for me. Remastersys, Pinguy, Customizer, Cubic - They have all got me somewhere close over the years.
I am testing using VirtualBox 6.1 on a EFI Ubuntu 18.04 host. When EFI is enabled in VitualBox, it just boots to EFI shell & I was pretty stuck from there. I eventually discovered that to get the ISO to boot in EFI, I have to enter a couple of EFI Shell commands:
FS0:
cd isolinux
bootx64.efi

The remainder of my issues were all to do with Ubiquity Installer & the customised environment I was building.
I am not using any of the (u)buntu-desktop meta packages, rather I build a custom openbox environment. This meant ubiquity needed some files that are not listed as dependencies; tzdata & language-pack-en(-base).
I have spent a few weekends now debugging and I finally figured out the final Ubiquity issue causing the installer crash with my build;

- I did not include a ubiquity-slideshow package. After answering all the installer questions, it crashed after after "copying files" & at the beginning of "installing locales". If I simply waited at the "user creation" screen of Ubiquity, the "copying files" seems to run in the background. After waiting for disk activity to stop & proceeding with "user creation", the installer worked fine. This was discovered purely by chance.
- adding ubiquity-slideshow-xubuntu (chosen for the small file size) , Ubiquity no longer crashes at any point. Don't know why this is, but I don't care anymore now I know the workaround.

Now all I need to figure out is how to get EFI booting to work without entering shell commands - just like a (u)buntu Live CD.
My assumption is that EFI/BOOT/BOOTx64.EFI is a default that EFI is looking for?
